R/tm_g_butterfly.R

```diff
@@ -360,6 +354,43 @@ srv_g_butterfly <- function(id, data, filter_panel_api, reporter, dataname, labe
       ADSL <- data[["ADSL"]]() # nolint
       ANL <- data[[dataname]]() # nolint
 
+      teal::validate_has_data(ADSL, min_nrow = 0, msg = sprintf("%s Data is empty", "ADSL"))
+      teal::validate_has_data(ANL, min_nrow = 0, msg = sprintf("%s Data is empty", dataname))
+
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("category_var", shinyvalidate::sv_required(
+        message = "Category Variable is required"
+      ))
+      iv$add_rule("right_var", shinyvalidate::sv_required(
+        message = "Right Dichotomization Variable is required"
+      ))
+      iv$add_rule("left_var", shinyvalidate::sv_required(
+        message = "Left Dichotomization Variable is required"
+      ))
```

Comment:
So we should also enforce that there should be at most two variables selected then we can remove the hacky code needed to enforce this at the moment

![image](https://user-images.githubusercontent.com/15201933/206422143-9e286a40-9aef-41f5-99d7-71d9d6bee6bd.png)

Reply:
How is this hacky? This is a legitimate option to `selectizeInput`. Plus the flag is right in the input. Do we have to pass this through a validator?

Reply:
Personally I would much rather have the "you can only select 2 options" as a validator than the code which does this https://github.com/insightsengineering/teal.osprey/blob/main/R/tm_g_butterfly.R#L282-L319

Reply:
That's not what that does, though, I think.
https://github.com/insightsengineering/teal.osprey/blob/main/R/tm_g_butterfly.R#L190-L198

Reply:
Wonder what that does do then? And can we remove it...

Reply:
Presents choices, I think. In a very roundabout way, which I am sure is supported by some desired behavior.